### PR TITLE
Test various LogManager.ThrowExceptions states

### PR DESCRIPTION
**Pull request by AnthonySteele**
More variations on LogManager.ThrowExceptions states
Fix to the failure discussed in https://github.com/NLog/NLog/issues/2057#issuecomment-293625641

There is still an issue where .Net core doesn't seem to be loading the `nlog.config` file.
Fortunately that is the domain of nlog itself, not NLog.StructuredLogging.Json

### CODE REVIEW

**Comment by jplebre** [comment]
LGTM, ~what is this trying to fix?~
Nevermind just read above

File: src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj [hunk 1]
```diff
@@ -11,6 +11,14 @@
...
     <Reference Include="System" />
     <Reference Include="Microsoft.CSharp" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Remove="nlog.config" />
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="nlog.config">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
```
Comment (jplebre) [inline src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:21]: confused. won't these conflict? remove and copy to output directory?

File: src/NLog.StructuredLogging.Json.Tests/nlog.config [hunk 2]
```diff
@@ -3,25 +3,15 @@
...
-      layout="${human_readable_layout}"/>
+    <target xsi:type="Debugger" name="Debugger" layout="${human_readable_layout}"/>
+    <target xsi:type="Console" name="Console" layout="${human_readable_layout}"/>
   </targets>
-
   <rules>
-    <logger name="*" minlevel="Trace" writeTo="console"/>
-    <logger name="*" minlevel="Trace" writeTo="debugger"/>
+    <logger name="*" minlevel="Trace" writeTo="Console"/>
+    <logger name="*" minlevel="Trace" writeTo="Debugger"/>
```
Comment (jplebre) [inline src/NLog.StructuredLogging.Json.Tests/nlog.config]: ?

File: src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj (hunk 1 above)
Comment (AnthonySteele) [reply src/NLog.StructuredLogging.Json.Tests/NLog.StructuredLogging.Json.Tests.csproj:21]: I did that from Visual studio. I think that the effect is to "remove it from the default rule, and then add a special case rule". What we need to happen in to make sure that `nlog.config` is always copied to the output directory.

File: src/NLog.StructuredLogging.Json.Tests/nlog.config (hunk 2 above)
Comment (AnthonySteele) [reply src/NLog.StructuredLogging.Json.Tests/nlog.config]: I was trying to get it working on Travis and linux. it didn't have any effect, so I should probably revert that part

**Pull request merged**